tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/mailet/CalDavCollect.java

```diff
@@ -79,21 +83,45 @@ public void service(Mail mail) throws MessagingException {
     }
 
     private void handleCalendarInMail(byte[] json, Mail mail) {
-        mail.getRecipients()
-            .forEach(mailAddress -> {
-                try {
-                    davUserProvider.provide(Username.of(mailAddress.asString()))
-                        .flatMap(davUser -> synchronizeWithDavServer(json, davUser))
-                        .block();
-                } catch (Exception e) {
-                    LOGGER.error("Error while handling calendar in mail {} with recipient {}", mail.getName(), mailAddress.asString(), e);
-                }
-            });
+        String icalContent = extractIcalContent(json, mail.getName());
+
+        if (!icalContent.isEmpty()) {
+            mail.getRecipients()
+                .forEach(mailAddress -> {
+                    try {
+                        if (isRecipientAttendee(mailAddress, icalContent)) {
+                            davUserProvider.provide(Username.of(mailAddress.asString()))
+                                .flatMap(davUser -> synchronizeWithDavServer(json, davUser))
+                                .block();
+                        }
+                    } catch (Exception e) {
+                        LOGGER.error("Error while handling calendar in mail {} with recipient {}", mail.getName(), mailAddress.asString(), e);
+                    }
+                });
+        }
     }
 
     private Mono<Void> synchronizeWithDavServer(byte[] json, DavUser davUser) {
         return davClient.sendITIPRequest(davUser.username(),
             URI.create(CALENDAR_PATH + davUser.userId()),
             json);
     }
+
+    private String extractIcalContent(byte[] json, String mailName) {
+        try {
+            String jsonString = new String(json, StandardCharsets.UTF_8);
+
+            ObjectMapper mapper = new ObjectMapper();
+            JsonNode rootNode = mapper.readTree(jsonString);
+
+            return rootNode.get("ical").asText();
+        } catch (Exception e) {
+            LOGGER.error("Error while handling calendar in mail {}", mailName, e);
+            return "";
+        }
+    }
+
+    private boolean isRecipientAttendee(MailAddress mailAddress, String icalContent) {
+        return icalContent.toLowerCase().contains(mailAddress.getLocalPart().toLowerCase());
+    }
```

Comment:
We need to be smarter and handle calendar data as structured data using ICAL4J

Reply:
I find that a bit overkill IMO as we already parsed and processed the calendar with Ical4j earlier in the pipeline, feels like a lot of extra processing to make it smarter and just check if recipient is part of the attendee list?

But I will give a shot